### DISCUSSION TITLE
huge memory usage

### DISCUSSION THREAD

**Discussion by Jamesun921** [rustdesk/rustdesk#5839] [General]
### Bug Description

huge memory usage on windows host

### How to Reproduce

connect to windows host by rustdesk client, watching the windows tasks, there shows a huge memory usage about 1Gb

### Expected Behavior

low memory even when connected

### Operating system(s) on local side and remote side

android or win --> win

### RustDesk Version(s) on local side and remote side

1.2.3 -->1.2.3 nightly

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/12782761/454ee49a-8964-46f6-8d05-8840e242b09c)

### Additional Context
Became very low memory usage when disconnect,and go high when connect again.

_No response_

**Comment by 21pages** [comment]
try https://github.com/rustdesk/rustdesk/discussions/5264 and tell me the result please.
Could you tell your resulotion?

**Comment by Jamesun921** [comment]
> try #5264

My windows host just do not have GPU.

**Comment by rustdesk** [comment]
@kuse2001 can you try out 1.2.2, same memory usage?

**Comment by rustdesk** [comment]
The memory accumulate gradually?

**Comment by Jamesun921** [comment]
> @kuse2001 can you try out 1.2.2, same memory usage?

Yes, formerly I tried the newest released version,  same result.

I noticed when I connect from remote, the task got high memory right off

**Comment by rustdesk** [comment]
@21pages @fufesou show your memory usage, and tell the display resolution.

**Comment by 21pages** [comment]
installed: 1920*1080, 4 process, av1 about 170M, vp9 about 380M

**Comment by rustdesk** [comment]
memory usage is very normal on my host also.

**Comment by rustdesk** [comment]
have to say this happens on specific machine only

**Comment by Jamesun921** [comment]
My resolution : 3840* 2160 , encode auto vp9 , quality: speed first

**Comment by rustdesk** [comment]
So 1G vs 380M, 1920x1080 vs 3840x2160, making sense.

**Reply by Jamesun921** [reply]
Can it be optimized?

**Reply by Jamesun921** [reply]
![image](https://github.com/rustdesk/rustdesk/assets/12782761/ca535998-5fec-4796-811c-7b729f137bf1)
Compare to anydesk，both are connecting.

**Comment by Jamesun921** [comment]
Hosting on my mac mini 2,the memory is 215M when connected, which is acceptable.

**Reply by rustdesk** [reply]
can you check av1 codec on your windows host?

**Reply by Jamesun921** [reply]
338M on av1 mode, 4K，but quality become not good

**Reply by 21pages** [reply]
> but quality become not good

Laggy or vague? what about balanced quality?

**Reply by Jamesun921** [reply]
I have try many config mode. Result ： resolution is not the primary reason, even quality  . Encode mode is the major cause 。Currently，I use av1 ，4k,   memory drop to 300M-400M which  is something acceptable. 

av1 got more vague in auto-adapting mode compare to none-adapting mode , so I will avoid using it.

**Reply by 21pages** [reply]
Thanks for your feedback,  none-adapting mode, speed quality mode, AV1 is significantly  more vague than VP9 ?

**Reply by Jamesun921** [reply]
Not much difference in picture quality when it is none-adapting mode, AV1 mode is slightly whiter by a little bit. Speed /quality mode are the same result.
